### PR TITLE
StudentFeedbackQuestionSubmitPageUiTest failing on dev server #4176

### LINKED ISSUE #4176: StudentFeedbackQuestionSubmitPageUiTest failing on dev server
image

### PR DESCRIPTION
**Pull request by taniach**
Fixes #4176

### CODE REVIEW

**Comment by damithc** [comment]
@junhaoyap can you do this quickly? 
@taniach Jun Hao may have gone to sleep already. Get someone else to review?

**Comment by damithc** [comment]
If you can get the tests to pass, you can go ahead and merge if you confident of the fix.

**Comment by junhaoyap** [comment]
I was having dinner, I can help take a look

**Comment by kanghj** [comment]
It looks fine to me, but @junhaoyap can take a final look

File: src/main/webapp/js/statusMessage.js [hunk 1]
```diff
@@ -1,7 +1,13 @@
 $(document).ready(function() {
     var statusMessage = document.getElementById('statusMessage');
-    var navbarHeight = document.getElementsByClassName('navbar')[0].offsetHeight;
+    var navbarHeight = 0;
```
Comment (junhaoyap) [inline src/main/webapp/js/statusMessage.js]: could name it as navbarOffsetHeight but it's really not that important if we're trying to release it asap.

**Comment by junhaoyap** [comment]
Yupp looks good

File: src/main/webapp/js/statusMessage.js (hunk 1 above)
Comment (taniach) [inline src/main/webapp/js/statusMessage.js]: This one is the exact height of the navbar (51px)

**Comment by taniach** [comment]
@junhaoyap Can you help me run the tests if you are free? The tests are slow on my computer again.

File: src/main/webapp/js/statusMessage.js [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
     var extraPadding = 15;
+    
+    var navbar = document.getElementsByClassName('navbar')[0];
+    
+    if (navbar != null) {
+        navbarHeight = navbar.offsetHeight;
```
Comment (junhaoyap) [inline src/main/webapp/js/statusMessage.js]: @taniach look here, offsetHeight and height are different, offsetHeight takes into account border and padding whereas .height() [for jQuery] or .clientHeight [vanilla JavaScript] returns the exact height of the element sans border and padding so we might want to make a distinction between the two of them for clarity.

But again, considering the urgency of this fix this isn't that important.

**Comment by junhaoyap** [comment]
Sorry maybe @kanghj can help you run it, I'm still halfway through setting up Teammates on my new machine and it might take awhile more before I can run it for you even if I complete it now :\

**Comment by junhaoyap** [comment]
But no worries, in general if this is the only thing causing the failure, I'm rather confident this fix will work, it does the right check in vanilla JavaScript of whether the element exists (!= null) or in jQuery (.length > 0).

**Comment by damithc** [comment]
@taniach just run the affected two tests and a couple of other tests. No need to run the whole suite. I'll be doing that anyway.

**Comment by taniach** [comment]
I added extraPadding so that the status message is not too close to the navbar and it looks better.

**Comment by taniach** [comment]
@damithc ok

**Comment by junhaoyap** [comment]
Yeah in this case .offsetHeight returns the exact height of the navbar element because the navbar element (if I'm not mistake) does not have padding or borders.

Yupp I agree with the extra space you added because the precise calculations is causing the status message to be too close to the top of the browser.

**Comment by taniach** [comment]
Took me 1.5hrs to run the entire test suite

**Comment by junhaoyap** [comment]
That took oddly longer than it should..

**Comment by damithc** [comment]
In my case it seems slightly faster than before. The entire suite took almost 9 minutes (5 failed test cases). Earlier it used to take nearly 10 minutes.

**Pull request merged**